### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by kmahajan11**
Hi Andrew - Please review my commits to the develop branch. I added my templates to the template folder and the assets to the assets folder as well as an updated README.

### CODE REVIEW

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 1]
```diff
@@ -0,0 +1,1075 @@
...
+          CloudTrailLogGroupArn:
+            type: String
+            default:
+              Fn::Join:
+                  - ''
+                  - - 'arn:aws:logs:'
+                    - Ref: AWS::Region
+                    - ':'
+                    - Ref: AWS::AccountId
+                    - !Sub ':log-group:${CloudTrailLogGroup}:*'
```
Comment (andrew-glenn) [inline templates/aws-cis-systemsmanagerautomations.yml:943]:
```suggestion
            default: !Sub "arn:${AWS::Partition}:${AWS::Region}:${AWS::AccountId}:log-group:${CloudTrailLogGroup}:*"
```

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 2]
```diff
@@ -0,0 +1,1075 @@
...
+          CloudWatchRoleArn:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref CloudWatchRole
```
Comment (andrew-glenn) [inline templates/aws-cis-systemsmanagerautomations.yml:952]:
```suggestion
            default: !Sub "arn${AWS::Partition}:iam::${AWS::AccountId}:role/${CloudWatchRole}"
```

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 3]
```diff
@@ -0,0 +1,1075 @@
...
+      Name: Custom-EnableVPCFlowLogsCF
+      Content:
+        schemaVersion: '0.3'
+        assumeRole:
+          Fn::Join:
+            - ''
+            - - 'arn:aws:iam::'
+              - Ref: AWS::AccountId
+              - ':role/'
+              - !Ref AutomationAssumeRole
```
Comment (andrew-glenn) [inline templates/aws-cis-systemsmanagerautomations.yml:984]:
```suggestion
        assumeRole: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${AutomationAssumeRole}"
```

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 4]
```diff
@@ -0,0 +1,1075 @@
...
+          CloudWatchLogGroupArn:
+            type: String
+            default:
+              Fn::Join:
+                  - ''
+                  - - 'arn:aws:logs:'
+                    - Ref: AWS::Region
+                    - ':'
+                    - Ref: AWS::AccountId
+                    - !Sub ':log-group:${FlowLogsCloudWatchLogs}'
```
Comment (andrew-glenn) [inline templates/aws-cis-systemsmanagerautomations.yml:995]:
```suggestion
            default: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:log-group:${FlowLogsCloudWatchLogs}"
```

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 5]
```diff
@@ -0,0 +1,1075 @@
...
+          FlowLogRoleArn:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref FlowLogsRole
```
Comment (andrew-glenn) [inline templates/aws-cis-systemsmanagerautomations.yml:1004]:
```suggestion
            default: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${FlowLogsRole}"
```

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 6]
```diff
@@ -0,0 +1,1075 @@
...
+            type: String
+          AutomationAssumeRole:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
```
Comment (andrew-glenn) [inline templates/aws-cis-systemsmanagerautomations.yml:1018]:
```suggestion
            default: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${AutomationAssumeRole}"
```

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 7]
```diff
@@ -0,0 +1,1075 @@
...
+      Name: Custom-RestrictSecurityGroup
+      Content:
+        schemaVersion: '0.3'
+        assumeRole:
+          Fn::Join:
+            - ''
+            - - 'arn:aws:iam::'
+              - Ref: AWS::AccountId
+              - ':role/'
+              - !Ref AutomationAssumeRole
```
Comment (andrew-glenn) [inline templates/aws-cis-systemsmanagerautomations.yml:1049]:
```suggestion
        assumeRole: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${AutomationAssumeRole}"
```

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 8]
```diff
@@ -0,0 +1,1075 @@
...
+            default: '0.0.0.0/0'
+          AutomationAssumeRole:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
```
Comment (andrew-glenn) [inline templates/aws-cis-systemsmanagerautomations.yml:1064]:
```suggestion
            default: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${AutomationAssumeRole}"
```

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 9]
```diff
@@ -0,0 +1,1075 @@
...
+              Service:
+                - ssm.amazonaws.com
+                - events.amazonaws.com
+                - ec2.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/IAMFullAccess
+        - arn:aws:iam::aws:policy/AdministratorAccess
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: Removed the IAM Full Access. However Administrator Access is prudent to account for several current use cases that require write permissions on several services like s3, ec2, cloudtrail, kms, etc as well as to accomodate future use cases for CIS. If absolutely necessary I can revisit

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 10]
```diff
@@ -0,0 +1,1075 @@
...
+# Bucket Policy for CIS S3 Bucket. Restrict to allow access to only SSL transport.
+  S3LoggingBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref S3LoggingBucket
+      PolicyDocument:
+        Statement:
+          -
+            Action:
+              - "s3:*"
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml:157]: done--took away s3:* and explicitly added permissions specific to bucket logging

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 11]
```diff
@@ -0,0 +1,1075 @@
...
+          - Effect: Allow
+            Action:
+            - ssm:StartAutomationExecution
+            - logs:CreateLogGroup
+            - logs:CreateLogStream
+            - logs:PutLogEvents
+            - logs:DescribeLogGroups
+            - logs:DescribeLogStreams
+            - iam:PassRole
+            - ec2:*
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: removed it --wasn't needed

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 12]
```diff
@@ -0,0 +1,1075 @@
...
+                    - Ref: AWS::AccountId
+                    - ':key/'
+                    - !Ref KmsKeyId
+        mainSteps:
+          - name: EncryptCloudTrail
+            action: 'aws:executeScript'
+            inputs:
+              Runtime: python3.6
+              Handler: encrypttrail_handler
+              Script: "def encrypttrail_handler(events, context):\r\n  import boto3\r\n  cloudtrail = boto3.client('cloudtrail')\r\n\r\n  CloudTrailLogGroupArn = events['CloudTrailLogGroupArn']\r\n  CloudWatchRoleArn = events['CloudWatchRoleArn']\r\n  TrailName = events['TrailName']\r\n  KMSKeyArn = events['KMSKeyArn']\r\n\r\n  response = cloudtrail.update_trail(\r\n    Name=TrailName,\r\n    IncludeGlobalServiceEvents=True,\r\n    IsMultiRegionTrail=True,\r\n    EnableLogFileValidation=True,\r\n    CloudWatchLogsLogGroupArn=CloudTrailLogGroupArn,\r\n    CloudWatchLogsRoleArn=CloudWatchRoleArn,\r\n    KmsKeyId=KMSKeyArn\r\n  )   "
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml:379]: yes not possible in ssm automation document

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 4 above)
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 5 above)
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 6 above)
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 7 above)
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 8 above)
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 13]
```diff
@@ -0,0 +1,1075 @@
...
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
+        mainSteps:
+          - name: rotatebackingkey
+            action: 'aws:executeScript'
+            inputs:
+              Runtime: python3.6
+              Handler: rotatebackingkey_handler
+              Script: "def rotatebackingkey_handler(events, context):\r\n  import boto3\r\n  client = boto3.client('kms')\r\n\r\n  KMSKeyArn = events['KMSKeyArn']\r\n\r\n  response = client.enable_key_rotation(\r\n    KeyId=KMSKeyArn\r\n  )"
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml:410]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 14]
```diff
@@ -0,0 +1,1075 @@
...
+                                PasswordReusePrevention=24,
+                                RequireLowercaseCharacters=True,
+                                RequireNumbers=True,
+                                RequireSymbols=True,
+                                RequireUppercaseCharacters=True)
+              InputPayload:
+                AutomationAssumeRole: '{{AutomationAssumeRole}}'
+
+# -------------------------------------------------------------------------------------------------------------------------------------------------------
+# CIS AWS Foundations Benchmark - 1.3-1.4 – Disable and Rotate IAM Access Key that is older than 90 days
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml:492]: yes sure will do that in the cis documentation/deployment guide

File: templates/aws-cis-cloudwatchlogmetricfilters.yml [hunk 15]
```diff
@@ -0,0 +1,587 @@
...
+        Statement:
+          - Effect: Allow
+            Principal:
+            Principal:
+              AWS:
+                'Fn::Join':
+                  - ''
+                  - - 'arn:aws:iam::'
+                    - Ref: 'AWS::AccountId'
+                    - ':root'
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-cloudwatchlogmetricfilters.yml:216]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 16]
```diff
@@ -0,0 +1,1075 @@
...
+                  except Exception as e:
+                     print(e)
+                     raise
+              InputPayload:
+                AutomationAssumeRole: '{{AutomationAssumeRole}}'
+                username: '{{username}}'
+
+
+# -------------------------------------------------------------------------------------------------------------------------------------------------------
+# CIS AWS Foundations Benchmark - 1.12 – Deactivate Root Account IAM Access Key
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml:547]: yes sure will do that in the cis deployment guide. Can you email me a copy of the draft..:-)

File: templates/aws-cis-cloudwatchlogmetricfilters.yml [hunk 17]
```diff
@@ -0,0 +1,587 @@
...
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AWSSupportAccess
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-cloudwatchlogmetricfilters.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 18]
```diff
@@ -0,0 +1,1075 @@
...
+Outputs:
+  FlowLogsRoleArn:
+    Description: Arn for FlowsLogsRole
+    Value: 
+      Fn::Join:
+        - ''
+        - - 'arn:aws:iam::'
+          - Ref: AWS::AccountId
+          - ':role/'
+          - !Ref FlowLogsRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 19]
```diff
@@ -0,0 +1,1075 @@
...
+  CloudTrailLogGroupArn:
+    Description: Arn for CloudTrail CloudWatch Logs
+    Value:
+      Fn::Join:
+          - ''
+          - - 'arn:aws:logs:'
+            - Ref: AWS::Region
+            - ':'
+            - Ref: AWS::AccountId
+            - !Sub ':log-group:${CloudTrailLogGroup}:*'
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 20]
```diff
@@ -0,0 +1,1075 @@
...
+      Name: CISCloudTrail
+  CloudWatchRoleArn:
+    Description: Arn for CloudTrail CloudWatch IAM Role 
+    Value:
+      Fn::Join:
+        - ''
+        - - 'arn:aws:iam::'
+          - Ref: AWS::AccountId
+          - ':role/'
+          - !Ref CloudWatchRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 21]
```diff
@@ -0,0 +1,1075 @@
...
+    Description: Arn for KMS CMK
+    Value:
+      Fn::Join:
+        - ''
+        - - 'arn:aws:kms:'
+          - Ref: AWS::Region
+          - ':'
+          - Ref: AWS::AccountId
+          - ':key/'
+          - !Ref KmsKeyId
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 22]
```diff
@@ -0,0 +1,1075 @@
...
+    Properties:
+      Bucket: !Ref CISS3CloudTrailBucket
+      PolicyDocument:
+        Statement:
+          -
+            Action:
+              - "s3:*"
+            Effect: "Allow"
+            Resource:
+              - !Sub arn:aws:s3:::${CISS3CloudTrailBucket}
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 23]
```diff
@@ -0,0 +1,1075 @@
...
+            Principal:
+              AWS:
+              - !Ref AWS::AccountId
+          -
+            Action:
+              - "s3:*"
+            Effect: "Allow"
+            Resource:
+              - !Sub arn:aws:s3:::${CISS3CloudTrailBucket}
+              - !Sub arn:aws:s3:::${CISS3CloudTrailBucket}/*
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 24]
```diff
@@ -0,0 +1,1075 @@
...
+            Principal:
+              Service:
+              - cloudtrail.amazonaws.com
+          -
+            Effect: Deny
+            Principal: "*"
+            Action: "*"
+            Resource:
+              - !Sub arn:aws:s3:::${CISS3CloudTrailBucket}
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 25]
```diff
@@ -0,0 +1,1075 @@
...
+    Properties:
+      Bucket: !Ref S3LoggingBucket
+      PolicyDocument:
+        Statement:
+          -
+            Action:
+              - "s3:*"
+            Effect: "Allow"
+            Resource:
+              - !Sub arn:aws:s3:::${S3LoggingBucket}
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 26]
```diff
@@ -0,0 +1,1075 @@
...
+              - !Sub arn:aws:s3:::${S3LoggingBucket}/*
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 27]
```diff
@@ -0,0 +1,1075 @@
...
+            Principal:
+              AWS:
+              - !Ref AWS::AccountId
+          -
+            Effect: Deny
+            Principal: "*"
+            Action: "*"
+            Resource:
+              - !Sub arn:aws:s3:::${S3LoggingBucket}
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 28]
```diff
@@ -0,0 +1,1075 @@
...
+              - !Sub arn:aws:s3:::${S3LoggingBucket}/*
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 29]
```diff
@@ -0,0 +1,1075 @@
...
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS:
+                'Fn::Join':
+                  - ''
+                  - - 'arn:aws:iam::'
+                    - Ref: 'AWS::AccountId'
+                    - ':root'
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 30]
```diff
@@ -0,0 +1,1075 @@
...
+      Name: Custom-CloudTrailEncryptionCF
+      Content:
+        schemaVersion: '0.3'
+        assumeRole:
+          Fn::Join:
+            - ''
+            - - 'arn:aws:iam::'
+              - Ref: AWS::AccountId
+              - ':role/'
+              - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 31]
```diff
@@ -0,0 +1,1075 @@
...
+        parameters:
+          AutomationAssumeRole:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 32]
```diff
@@ -0,0 +1,1075 @@
...
+          CloudTrailLogGroupArn:
+            type: String
+            default:
+              Fn::Join:
+                  - ''
+                  - - 'arn:aws:logs:'
+                    - Ref: AWS::Region
+                    - ':'
+                    - Ref: AWS::AccountId
+                    - !Sub ':log-group:${CloudTrailLogGroup}:*'
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 33]
```diff
@@ -0,0 +1,1075 @@
...
+          CloudWatchRoleArn:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref CloudWatchRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 34]
```diff
@@ -0,0 +1,1075 @@
...
+            type: String
+            default:
+              Fn::Join:
+                  - ''
+                  - - 'arn:aws:kms:'
+                    - Ref: AWS::Region
+                    - ':'
+                    - Ref: AWS::AccountId
+                    - ':key/'
+                    - !Ref KmsKeyId
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 35]
```diff
@@ -0,0 +1,1075 @@
...
+            type: String
+          AutomationAssumeRole:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 36]
```diff
@@ -0,0 +1,1075 @@
...
+      Content:
+        description: CIS 2.4 – Ensure CloudTrail trails are integrated with Amazon CloudWatch Logs
+        schemaVersion: '0.3'
+        assumeRole:
+          Fn::Join:
+            - ''
+            - - 'arn:aws:iam::'
+              - Ref: AWS::AccountId
+              - ':role/'
+              - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 37]
```diff
@@ -0,0 +1,1075 @@
...
+        parameters:
+          AutomationAssumeRole:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 38]
```diff
@@ -0,0 +1,1075 @@
...
+          CloudTrailLogGroupArn:
+            type: String
+            default:
+              Fn::Join:
+                  - ''
+                  - - 'arn:aws:logs:'
+                    - Ref: AWS::Region
+                    - ':'
+                    - Ref: AWS::AccountId
+                    - !Sub ':log-group:${CloudTrailLogGroup}:*'
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 39]
```diff
@@ -0,0 +1,1075 @@
...
+      Name: Custom-IAMPasswordUpdateCF
+      Content:
+        schemaVersion: '0.3'
+        assumeRole:
+          Fn::Join:
+            - ''
+            - - 'arn:aws:iam::'
+              - Ref: AWS::AccountId
+              - ':role/'
+              - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 40]
```diff
@@ -0,0 +1,1075 @@
...
+            default: 'cisadmin'
+          AutomationAssumeRole:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 41]
```diff
@@ -0,0 +1,1075 @@
...
+      Name: Custom-DeactivateRootIAMAccessKeyCF
+      Content:
+        schemaVersion: '0.3'
+        assumeRole:
+          Fn::Join:
+            - ''
+            - - 'arn:aws:iam::'
+              - Ref: AWS::AccountId
+              - ':role/'
+              - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 42]
```diff
@@ -0,0 +1,1075 @@
...
+      Name: Custom-IAMUserPolicyDetachCF
+      Content:
+        schemaVersion: '0.3'
+        assumeRole:
+          Fn::Join:
+            - ''
+            - - 'arn:aws:iam::'
+              - Ref: AWS::AccountId
+              - ':role/'
+              - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 43]
```diff
@@ -0,0 +1,1075 @@
...
+            default: '0123'
+          AutomationAssumeRole:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 44]
```diff
@@ -0,0 +1,1075 @@
...
+      Name: Custom-IAMFullAdminPolicyDetachCF
+      Content:
+        schemaVersion: '0.3'
+        assumeRole:
+          Fn::Join:
+            - ''
+            - - 'arn:aws:iam::'
+              - Ref: AWS::AccountId
+              - ':role/'
+              - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 1 above)
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 3 above)
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml (hunk 2 above)
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

File: templates/aws-cis-systemsmanagerautomations.yml [hunk 45]
```diff
@@ -0,0 +1,1075 @@
...
+            default: 'cisadmin'
+          AutomationAssumeRole:
+            type: String
+            default:
+              Fn::Join:
+                - ''
+                - - 'arn:aws:iam::'
+                  - Ref: AWS::AccountId
+                  - ':role/'
+                  - !Ref AutomationAssumeRole
```
[the remark this answers is not in the document]
Comment (kmahajan11) [reply templates/aws-cis-systemsmanagerautomations.yml]: done

**Pull request merged**